modules/utxo-lib/test/integration_local_rpc/generate/RpcClient.ts

```diff
@@ -16,40 +16,98 @@ function sleep(millis: number): Promise<void> {
   });
 }
 
+export class RpcError extends Error {
+  constructor(public rpcError: { code: number; message: string }) {
+    super(`RPC error: ${rpcError.message} (code=${rpcError.code})`);
+  }
+
+  static isRpcErrorWithCode(e: Error, code: number): boolean {
+    return e instanceof RpcError && e.rpcError.code === code;
+  }
+}
+
+type NetworkInfo = { subversion: string };
+
+const BITCOIN_CORE_22_99 = '/Satoshi:22.99.0/';
+
 export class RpcClient {
   id = 0;
 
-  constructor(private network: Network, private url: string) {}
+  constructor(protected network: Network, protected url: string, protected networkInfo?: NetworkInfo) {}
+
+  /**
+   * Poor man's Bluebird.map(arr, f, { concurrency })
+   * Processes promises in batches of 16
+   *
+   * @param arr
+   * @param f
+   * @param [concurrency=8]
+   */
+  static async parallelMap<S, T>(
+    arr: S[],
+    f: (S, i: number) => Promise<T>,
+    { concurrency }: { concurrency: number } = { concurrency: 16 }
+  ): Promise<T[]> {
+    const rest: S[] = arr.splice(concurrency);
+    const result = await Promise.all(arr.map((v, i) => f(v, i)));
+    if (rest.length) {
+      return [...result, ...(await this.parallelMap(rest, f))];
+    }
+    return result;
+  }
+
+  protected getUrl(): string {
+    return this.url;
+  }
 
   async exec<T>(method: string, ...params: unknown[]): Promise<T> {
     try {
-      debug('<', method, params);
-      const response = await axios.post(this.url, {
+      debug('>', this.getUrl(), method, params);
+      const response = await axios.post(this.getUrl(), {
         jsonrpc: '1.0',
         method,
         params,
         id: `${this.id++}`,
       });
-      debug('>', response.data.result);
+      if (method === 'generate' || method === 'generatetoaddress') {
+        debug('<', '[...]');
+      } else {
+        debug('<', response.data.result);
+      }
       return response.data.result;
     } catch (e) {
       if (e.isAxiosError && e.response) {
         e = e as AxiosError;
-        const { error } = e.response.data;
-        const { code, message } = error;
-        throw new Error(`RPC error: ${message} (code=${code})`);
+        debug('< ERROR', e.response.statusText, e.response.data);
+        e = e as AxiosError;
+        const { error = {} } = e.response.data;
+        throw new RpcError(error);
       }
 
       throw e;
     }
   }
 
+  requiresWalletPath(): boolean {
+    if (!this.networkInfo) {
+      throw new Error(`networkInfo must be set`);
+    }
+    return this.networkInfo.subversion === BITCOIN_CORE_22_99;
```

Comment:
Why the 99 version here and not 22.0.0 as well?

Reply:
the `22.0.0` version does not require the wallet path in the RPC URL